expr/parse.go

```diff
@@ -476,3 +476,38 @@ func aggKey(serie models.Series, nodes []expr) string {
 	}
 	return strings.Join(name, ".")
 }
+
+// filterNodesByPositions removes the node at the given position(s)
+func filterNodesByPositions(serie models.Series, nodes []expr) (string, error) {
+	metric := extractMetric(serie.Target)
+	if len(metric) == 0 {
+		return "", fmt.Errorf("invalid metric name")
+	}
+	// Trim off tags (if they are there) and split on '.'
+	parts := strings.Split(strings.SplitN(metric, ";", 2)[0], ".")
+	name := parts[:0]
+
+	n := 0
+	for i, p := range parts {
+		// if we have already reached the end of the nodes to filter, just add all remaining parts to name
+		if n >= len(nodes) {
+			name = append(name, p)
+			continue
+		}
+
+		// we don't perform a check of ettype here. This should be done before calling this function
```

Comment:
assumptions like this are better written in the function documentation such that it's more obvious to the caller